runtime/lua/vim/iter.lua

```diff
@@ -112,6 +112,35 @@ local function sanitize(t)
   return t
 end
 
+---Flattens a single list-like table. Errors if it attempts to flatten a
+---dict-like table
+---@param v table table which should be flattened
+---@param max_depth number depth to which the table should be flattened
+---@param initial_depth number|nil (internal) starting depth (defaulted to 0)
+---@param result table|nil (internal) flattened in each iteration (defaulted to {})
```

Comment:
What does `(internal)` mean? That's not a convention that I've seen used in the Neovim code base.

Reply:
this is a helper function that recursively flattens a table, and these variables are only used at higher levels of recursion - not on the initial call. that's what I meant here - I can remove this if it's confusing